File: ghost-coach-be/src/main/java/com/playmotech/ghostcoach/auth/dto/RegisterRequest.java
package com.playmotech.ghostcoach.auth.dto;

import com.playmotech.ghostcoach.user.ExperienceLevel;
import com.playmotech.ghostcoach.user.Sport;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@ValidSportPosition
public record RegisterRequest(
        @NotBlank @Email
        String email,

        @NotBlank @Size(min = 8, max = 100)
        String password,

        @NotBlank @Size(min = 2, max = 100)
        String fullName,

        @NotNull
        Sport sport,

        @NotBlank @Size(max = 50)
        String position,

        @NotNull
        ExperienceLevel experienceLevel
) {}

File: ghost-coach-mobile/lib/data/model/auth/register_request.dart
import 'package:ghost_coach/data/model/enums/experience_level.dart';
import 'package:ghost_coach/data/model/enums/sport.dart';

class RegisterRequest {
  const RegisterRequest({
    required this.email,
    required this.password,
    required this.fullName,
    required this.sport,
    required this.position,
    required this.experienceLevel,
  });

  final String email;
  final String password;
  final String fullName;
  final Sport sport;
  final String position;
  final ExperienceLevel experienceLevel;

  Map<String, dynamic> toJson() => <String, dynamic>{
        'email': email,
        'password': password,
        'fullName': fullName,
        'sport': sport.wire,
        'position': position,
        'experienceLevel': experienceLevel.wire,
      };
}
